File: go-client-sdk/pkg/models/operations/searchcrmcontacts.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type SearchCrmContactsQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type SearchCrmContactsRequestBodyFilters struct {
	CreatedTime  *shared.DateFilter   `json:"createdTime,omitempty"`
	Email        *shared.StringFilter `json:"email,omitempty"`
	FirstName    *shared.StringFilter `json:"firstName,omitempty"`
	ID           *shared.StringFilter `json:"id,omitempty"`
	JobTitle     *shared.StringFilter `json:"jobTitle,omitempty"`
	LastName     *shared.StringFilter `json:"lastName,omitempty"`
	MobilePhone  *shared.StringFilter `json:"mobilePhone,omitempty"`
	ModifiedTime *shared.DateFilter   `json:"modifiedTime,omitempty"`
	NativeID     *shared.StringFilter `json:"nativeId,omitempty"`
	Phone        *shared.StringFilter `json:"phone,omitempty"`
}

type SearchCrmContactsRequestBody struct {
	Filters *SearchCrmContactsRequestBodyFilters `json:"filters,omitempty"`
}

type SearchCrmContactsResponseBody struct {
	Contacts       []shared.Contact `json:"contacts,omitempty"`
	NextPageCursor *string          `json:"nextPageCursor,omitempty"`
}

type SearchCrmContactsRequest struct {
	QueryParams SearchCrmContactsQueryParams
	Request     *SearchCrmContactsRequestBody `request:"mediaType=application/json"`
}

type SearchCrmContactsResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *SearchCrmContactsResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/searchcrmcontacts.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class SearchCrmContactsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class SearchCrmContactsRequestBodyFilters extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=email" })
  email?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=firstName" })
  firstName?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=jobTitle" })
  jobTitle?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=lastName" })
  lastName?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=mobilePhone" })
  mobilePhone?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=modifiedTime" })
  modifiedTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=nativeId" })
  nativeId?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=phone" })
  phone?: shared.StringFilter;
}


export class SearchCrmContactsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=filters" })
  filters?: SearchCrmContactsRequestBodyFilters;
}


export class SearchCrmContactsResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contacts", elemType: shared.Contact })
  contacts?: shared.Contact[];

  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;
}


export class SearchCrmContactsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: SearchCrmContactsQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: SearchCrmContactsRequestBody;
}


export class SearchCrmContactsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: SearchCrmContactsResponseBody;
}
